dev/tasks/python-wheels/travis.manylinux2010.yml

```diff
@@ -0,0 +1,82 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+os: linux
+
+services:
+  - docker
+
+# Update docker to support newer docker-compose versions
+addons:
+  apt:
+    packages:
+      - docker-ce
+
+# don't build twice
+if: tag IS blank
+
+env:
+  global:
+    - PLAT=x86_64
+    - TRAVIS_TAG={{ task.tag }}
+
+before_script:
+  - sudo pip install -U docker-compose
+
+script:
+  - git clone --no-checkout {{ arrow.remote }} arrow
+  - git -C arrow fetch -t {{ arrow.remote }} {{ arrow.branch }}
+  - git -C arrow checkout {{ arrow.head }}
+  - mkdir -p dist
+
+  # build wheel
+  - pushd arrow
+
+  # Pull testing resources
+  - git submodule init
+  - git submodule update
+
+  # don't attempt to build it, because the base image by pypa is updated
+  # regularly and would cause undeterministic builds
+  - docker-compose pull python-manylinux2010
+  - docker-compose run
+      -e SETUPTOOLS_SCM_PRETEND_VERSION={{ arrow.no_rc_version }}
+      -e PYTHON_VERSION="{{ python_version }}"
+      -e UNICODE_WIDTH="{{ unicode_width }}"
+      python-manylinux2010
+  - popd
+
+  # test on multiple distributions
+  {%- for image in test_docker_images %}
+  - docker run -it --shm-size 2G --volume $(pwd)/arrow:/arrow {{ image }}
+      /arrow/dev/tasks/python-wheels/manylinux2010-test.sh
```

Comment:
@pitrou I suggest you to add a `script` parameter to the [params list](https://github.com/apache/arrow/pull/4675/files#diff-0abe5ae3589de012350559065a4194a6R163) then `{{ script }}` jinja variable will contain the script's name (this way you can avoid the duplication of the travis template).

Reply:
Just see that the the parameter needs to be replaced in the docker-compose commands as well, still you can use something like `{{ manylinux_tag }}`

Reply:
I don't know. On the one hand, it would reduce duplication now. On the other hand, the builds may differ in the future. @wesm What do you think?

Reply:
Ideally We should not edit / manually maintain the CI templates in the future, offloading the complexity to other scripts like archery or docker containers and using generic templates.

Reply:
In this particular case I'm not sure. I'm in favor of efforts to reuse code in general